### PR TITLE
Draft a release post for v3.7.3

### PR DESCRIPTION
**Pull request by ashmaroli**
Release post for a patch release that fixes a bug introduced in `v3.7.0`
(Other bug-fixes should be released as part of `v3.8.0` IMHO)

### Requirements:
- [x] Merge #6791 to `master` branch
- [x] Backport #6791 to `3.7-stable` branch

### CODE REVIEW

File: docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+title: 'Jekyll 3.7.3 Released'
+date: 2018-02-25 13:02:08 +0530
+author: ashmaroli
+version: 3.7.3
+categories: [release]
+---
+
+Hello Jekyllers!! :wave:
+We're pleased to announce the release of `v3.7.3` which fixes a bug one might encounter while using `Jekyll - 3.7.x` along with a
+jekyll plugin that in turn uses the `I18n` library.
```
Comment (oe) [inline docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown]: capitalize `Jekyll` here

File: docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+When [v3.7.0]({% link _posts/2018-01-02-jekyll-3-7-0-released.md %}) enhanced our `slugify` filter with a `latin` option, we also
+hardcoded a default fallback locale for the `I18n` library to avoid an exception raised in the event the library fails to find
+any locale. This led to issues with third-party i18n plugins for Jekyll especially since default locale that got assigned before
```
Comment (oe) [inline docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown]: comma before `especially`

Comment (oe) [reply docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown]: `the default locale`

File: docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+the plugin was loaded, and irrespective of whether the `slugify` filter was used.
```
Comment (oe) [inline docs/_posts/2018-02-25-jekyll-3-7-3-released.markdown]: remove `and` here

**Comment by oe** [comment]
Backported #6791 to https://github.com/jekyll/jekyll/tree/3.7-stable

**Comment by oe** [comment]
Going to release the gem and then merge this 🎉

**Comment by ashmaroli** [comment]
> Going to release the gem

I'm game as long as @pathawks is fine with this too..

**Comment by pathawks** [comment]
Thanks for your work on this, @ashmaroli

**Comment by oe** [comment]
@jekyllbot: merge +docs

**Pull request merged**